apps/archive/common.py

```diff
@@ -204,7 +213,11 @@ def on_create_item(docs, repo_type=ARCHIVE):
             # set the source for the article
             set_default_source(doc)
 
-        if 'profile' not in doc and app.config.get('DEFAULT_CONTENT_TYPE', None):
+        ignore_profiles = DEFAULT_PROFILES.copy()
+        ignore_profiles.add(None)
+
+        if doc.get('profile') in ignore_profiles and doc.get('type') == 'text' and \
+                app.config.get('DEFAULT_CONTENT_TYPE', None):
             doc['profile'] = app.config.get('DEFAULT_CONTENT_TYPE', None)
```

Comment:
just to make it easier to understand: `app.config['DEFAULT_CONTENT_TYPE']`
Since at this point `DEFAULT_CONTENT_TYPE` is set anyway.